CHANGELOG.md: 3.1.14 breaking change (#2344)
rack/rack: CHANGELOG.md

## [3.1.14] - 2025-05-06

> :warning: **This release includes a breaking change.**
> It introduces two limits `bytesize_limit` and `params_limit`. Which limits "total query size" and "total number of query parameters".
> 

Reviewer: ```suggestion
```

Reviewer 2: ```suggestion
:warning: **This release includes a security fix that may cause certain routes in previously working applications to fail if query parameters exceed 4,096 in count or 4 MB in total size.**
```

### Security

> - [CVE-2025-46727](https://github.com/rack/rack/security/advisories/GHSA-gjh7-p2fx-99vx) Unbounded parameter parsing in `Rack::QueryParser` can lead to memory exhaustion.

Reviewer: ```suggestion
- [CVE-2025-46727](https://github.com/rack/rack/security/advisories/GHSA-gjh7-p2fx-99vx) Unbounded parameter parsing in `Rack::QueryParser` can lead to memory exhaustion. This was addressed by introducing two limits, `bytesize_limit` (total query size) and `params_limit` (total number of query parameters). Applications that expect to parse very large queries or queries with a very large number of parameters may need to increase the default limits.
```

Reviewer: For the vast majority of applications, it's not a breaking change, as the applications are under the default limits. For a small percentage of applications that expect requests that exceed the default limits, it is breaking.

We previously weren't mentioning the limits and that they may need to be updated at all in CHANGELOG. I do think it is worth mentioning the limits and that some applications may need to increase them, but in the same section as the security change.

My changes are deliberate and considered. If you think it should be displayed differently, I have no objections. Can you please either merge or close as **you** see fit?

Reviewer 2: Sure, no problem.
